### PR TITLE
Refactor Button Styles

### PR DESCRIPTION
**Pull request by 0916dhkim**
## Description

- Remove `styled-components` usage.
- Use `StyleSheet` module from React-Native instead.
- Combine all style related props into single `style` prop.
- Update all storyboards which use `Button` component.

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/dooboo-ui/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by 0916dhkim** [comment]
Snapshots are failing. Should I update the jest snapshots?

**Comment by JeffGuKang** [comment]
> Snapshots are failing. Should I update the jest snapshots?

Thanks for PR. It will be tested again after solving #337.

**Comment by 0916dhkim** [comment]
@JeffGuKang I rebased this PR on top of #349 . The number of failing snapshots is reduced from 4 to 1.
Here's the diff of the failing snapshot.
```diff
- Snapshot  - 0
+ Received  + 4

@@ -3,10 +3,11 @@
      Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
+       undefined,
      ]
    }
  >
    <
      activeOpacity={0.7}
@@ -25,20 +26,23 @@
            "paddingRight": 16,
            "width": 320,
          },
          undefined,
          false,
+         undefined,
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#069ccd",
              "fontSize": 14,
            },
+           undefined,
+           undefined,
          ]
        }
      />
    </>
  </View>
```

There are 4 additional `undefined` elements inside `style` props, which I think is harmless.

**Comment by 0916dhkim** [comment]
@hyochan Can you review this PR? I'm also refactoring `EditText` component in a similar fashion, so I'd like some feedback.

**Comment by 0916dhkim** [comment]
Synchronized `Button` documentation with actual source code.
https://github.com/dooboolab/dooboo-ui/pull/348/commits/9135fc3808b370fead0fe84af8e3f9f45748b07c
@JeffGuKang  This PR will resolve #350 .

File: main/Button/index.tsx [hunk 1]
```diff
@@ -1,147 +1,139 @@
...
-
-const StyledDisabledText = styled(StyledText)`
-  color: #969696;
-`;
-
-interface ChildStyle {
-  style?: ViewStyle;
-  textStyle?: TextStyle;
-}
+const defaultStyles = StyleSheet.create({
```
Comment (JeffGuKang) [inline main/Button/index.tsx:13]: It will be better to move `StyleSheet` to just before `export default`.

File: main/Button/index.tsx [hunk 2]
```diff
@@ -1,147 +1,139 @@
...
+    backgroundColor: '#cccccc',
+    borderColor: 'rgb(200, 200, 200)',
+  },
+  disabledText: {
+    color: '#969696',
+  },
+});
 
-type ColorType = 'decision' | 'idea' | 'emotion' | 'benefit' | 'facts' | 'criticism';
+type StylesType = Partial<StyleSheet.NamedStyles<typeof defaultStyles>>;
```
Comment (JeffGuKang) [inline main/Button/index.tsx:54]: Clever!

**Review by JeffGuKang — APPROVED** [review]

**Comment by JeffGuKang** [comment]
@hyochan There are big changes in `Button`. @0916dhkim grouped styles props into one style prop that can receive various styles. How about this way?

**Comment by hyochan** [comment]
> @hyochan There are big changes in `Button`. @0916dhkim grouped styles props into one style prop that can receive various styles. How about this way?

Looks good to me!

**Pull request merged**